### PR TITLE
Pin stable rust

### LINKED ISSUE #3257: Add pinned version of the Rust toolchain to the node
We should add a `rust-toolchain.toml` file to the root of the repo with the following contents:
```toml
[toolchain]
channel = "1.63"
```
The Makefile `setup-rs` target will also need to be updated to replace both instances of `stable` with the value read from the `rust-toolchain.toml` file using something like
```
PINNED_STABLE  := $(shell grep -Eo '[0-9]+\.[0-9]+\.?[0-9]*' rust-toolchain.toml)
```

### PR DESCRIPTION
**Pull request by darthsiroftardis**
CHANGELOG:

- Added a rust-toolchain.toml pinning rust stable 1.63

Closes #3257

### CODE REVIEW

**Review by Fraser999 — APPROVED** [review]

**Comment by rafal-ch** [comment]
Can you explain in the description **why** we need that? The connected ticket only says what we should do, not why.

File: .github/workflows/lints.yml [hunk 1]
```diff
@@ -36,11 +36,15 @@ jobs:
         id: nightly-toolchain
         run: echo "::set-output name=version::$(cat smart_contracts/rust-toolchain)"
 
+      - name: Get stable from rust-toolchain.toml
+        id: stable-toolchain
+        run: echo "::set-output name=version::$(grep -Eo '[0-9]+\.[0-9]+\.?[0-9]*' rust-toolchain.toml)"
```
Comment (rafal-ch) [inline .github/workflows/lints.yml]: Could you consider using:
```
sed -nr 's/channel\s+=\s+\"(.*)\"/\1/p' rust-toolchain.toml
```

This way we'll also support toolchain specifiers such as
```
[toolchain]
channel = "stable"
```
and possibly all other combinations.

Ultimately, some toml parser could come in handy (something similar to `jq` for json).

File: Makefile [hunk 2]
```diff
@@ -4,6 +4,7 @@ RUSTUP = $(or $(shell which rustup), $(HOME)/.cargo/bin/rustup)
 NPM    = $(or $(shell which npm),    /usr/bin/npm)
 
 PINNED_NIGHTLY := $(shell cat smart_contracts/rust-toolchain)
+PINNED_STABLE  := $(shell grep -Eo '[0-9]+\.[0-9]+\.?[0-9]*' rust-toolchain.toml)
```
Comment (rafal-ch) [inline Makefile]: Please consider leveraging `sed` here as well.

File: .github/workflows/lints.yml (hunk 1 above)
Comment (darthsiroftardis) [reply .github/workflows/lints.yml]: No worries, I was going with the recommendation in the ticket, but can do for sure yeah!

File: Makefile (hunk 2 above)
Comment (darthsiroftardis) [reply Makefile]: Applied suggestion

File: .github/workflows/lints.yml (hunk 1 above)
Comment (darthsiroftardis) [reply .github/workflows/lints.yml]: Applied suggestion

**Review by Fraser999 — APPROVED** [review]

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Comment by goral09** [comment]
bors cancel

**Pull request merged**